sdk/inc/azure/core/az_json.h

```diff
@@ -792,6 +792,27 @@ AZ_NODISCARD az_result az_json_reader_next_token(az_json_reader* ref_json_reader
  */
 AZ_NODISCARD az_result az_json_reader_skip_children(az_json_reader* ref_json_reader);
 
+/**
+ * @brief Gets the JSON token's string after unescaping it, if required.
+ *
+ * @param[in] json_string The #az_span that contains the string to be unnescaped.
+ * @param destination Pointer to the buffer that will contain the oputput.
+ * @param[in] destination_max_size The maximum available space within the buffer referred to by
+ * \p destination.
+ * @param[out] out_string_length __[nullable]__ Contains the number of bytes written to the
+ * \p destination which denote the length of the unescaped string. If `NULL` is passed, the
+ * \p parameter is ignored.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The string is returned.
+ * @retval #AZ_ERROR_NOT_ENOUGH_SPACE \p destination does not have enough size.
+ */
+AZ_NODISCARD az_result az_json_string_unescape(
+    az_span json_string,
+    char* destination,
```

Comment:
@ahsonkhan just curious: why was this originally created as a `char*` and `out_string_length` instead of an `az_span` ?
Is it guaranteed in the original `az_json_token_get_string()` API that the output is a `NULL`-terminated string?

Reply:
The API signature is the one that came from the IOT team , it is in the issue associated with the PR. While i agree with the point i suspect that there might be some other reason for it

Reply:
> Is it guaranteed in the original `az_json_token_get_string()` API that the output is a `NULL`-terminated string?

Yes.